WordPress/Tests/WP/UseCapabilitiesNotRolesUnitTest.inc

```diff
@@ -0,0 +1,47 @@
+<?php
+add_dashboard_page(
+	'page_title',
+	'menu_title',
+	'super_admin', // Error.
+	'menu_slug',
+	'function'
+);
+add_posts_page( 'page_title', 'menu_title', 'administrator', 'menu_slug', 'function' ); // Error.
+add_media_page( 'page_title', 'menu_title', 'editor', 'menu_slug', 'function' ); // Error.
+add_pages_page( 'page_title', 'menu_title', 'author', 'menu_slug', 'function' ); // Error.
+add_comments_page( 'page_title', 'menu_title', 'contributor', 'menu_slug', 'function' ); // Error.
+add_theme_page( 'page_title', $menu_title, 'subscriber', 'menu_slug', 'function' );  // Error.
+add_plugins_page( 'page_title', 'menu_title', 'super_admin', 'menu_slug', 'function' );  // Error.
+add_users_page( 'page_title', 'menu_title', 'administrator', 'menu_slug', 'function' );  // Error.
+add_management_page( 'page_title', 'menu_title', 'editor', 'menu_slug', 'function' );  // Error.
+add_options_page( 'page_title', 'menu_title', 'contributor', 'menu_slug', 'function' );  // Error.
+if ( current_user_can( 'super_admin' ) ) { } // Error.
+if( current_user_can_for_blog( '1', 'editor' ) ) { }  // Error.
+
+// @codingStandardsChangeSetting WordPress.WP.UseCapabilitiesNotRoles check_only_known_caps false
+add_menu_page( $pagetitle, $menu_title, $subscriber, 'handle', 'function', 'icon_url' ); // Warning.
+add_plugins_page( 'page_title', 'menu_title', $cap, 'menu_slug', 'function' );  // Warning.
+add_users_page( 'page_title', 'menu_title', 'xxx', 'menu_slug', 'function' ); // Warning.
+add_management_page( 'page_title', 'menu_title', 'xxx', 'menu_slug', 'function' ); // Warning.
+add_options_page( $pagetitle, $menu_title, $xxx, 'menu_slug', 'function' ); // Warning.
+add_menu_page( $pagetitle, 'menu_title', 'xxx', 'handle', 'function', 'icon_url' ); // Warning.
+add_utility_page(
+	'page_title', 'menu_title', 'super_admin', 'menu_slug', 'function', 'icon_url'  // Error.
+);
+add_submenu_page(
+	'parent_slug',
+	'page_title',
+	'menu_title',
+	$variable, // Warning.
+	'menu_slug',
+	'function'
+);
+if ( author_can( $post, $capability ) ) { } // Warning.
+if ( author_can( $post, 'FFFFF' ) ) { }  // Warning.
+if ( current_user_can( 'SSSSS' ) ) { } // Warning.
+if( current_user_can_for_blog( '3', 'EEEEEE' ) ) { }  // Warning.
+// @codingStandardsChangeSetting WordPress.WP.UseCapabilitiesNotRoles check_only_known_caps true
+
+// @codingStandardsChangeSetting WordPress.WP.UseCapabilitiesNotRoles custom_capabilities custom_cap
+if ( author_can( $post, 'custom_cap' ) ) { } // OK.
+if ( author_can( $post, 'read' ) ) { } // OK.
```

Comment:
Maybe also add:
```php
// @codingStandardsChangeSetting WordPress.WP.UseCapabilitiesNotRoles check_only_known_caps false
if ( author_can( $post, 'unknown_custom_cap' ) ) { } // Warning.
// @codingStandardsChangeSetting WordPress.WP.UseCapabilitiesNotRoles check_only_known_caps true
```
... to test that the warning is still thrown correctly even when custom capabilities have been registered.

A unit test with several custom caps would also not be a bad thing. Doesn't have to be a separate test, you could just change the settings line here (use a comma-delimited list without spaces) and test both custom caps.